### ISSUE TITLE
clustering

### ISSUE THREAD

**Issue by ahmadnassri** [Mashape/kong#651]
Discussion started in https://github.com/Mashape/kong/issues/15 need to define requirements for cluster awareness.

- cluster status API, similar to `/status` but for the whole cluster
- add remove nodes from cluster with API ?

**Comment by thefosk** [comment]
Supported commands in the first version will be:

* `join`
* `members`
* `keygen`
* `leave`
* `force-leave`

The configuration options available will be:

```yaml
######
## Cluster settings
cluster:

  ######
  ## Address to bind network listeners to.
  bind: "0.0.0.0:7946"

  ######
  ## Network interface to bind to. Can be used instead of bind if the interface is known but 
  ## not the address. If both are provided, then Kong verifies that the interface has the bind 
  ## address that is provided.
  iface: ""

  ######
  ## Address to bind the RPC listener.
  rpc-addr: "127.0.0.1:7373"

  ######
  ## Key for encrypting network traffic within Kong. Must be a base64-encoded 16-byte key.
  encrypt: "foo"
```

**Comment by ahmadnassri** [comment]
can we not have clustering be done through admin API, rather than yml/config?

seems to me it would be more powerful to dynamically setup rather to do on init.

**Comment by thefosk** [comment]
> can we not have clustering be done through admin API, rather than yml/config?

Yes. Very good point.

**Comment by Nijikokun** [comment]
> can we not have clustering be done through admin API, rather than yml/config?

:+1:

**Comment by ahmadnassri** [comment]
@thefosk can this be closed now with the features introduced in 0.6.0-RC?
